test/dfusionMultiSafes.js

```diff
@@ -165,52 +167,200 @@ contract("GnosisSafe", function(accounts) {
+      // Build orders
+      const orderTransaction = await buildOrders(
+        masterSafe.address,
+        bracketAddresses,
+        targetTokenId,
+        stableTokenId,
+        lowestLimit,
+        highestLimit
+      )
+      await execTransaction(masterSafe, lw, orderTransaction)
+
+      // Make transfers
+      const batchTransaction = await buildTransferApproveDepositFromOrders(
+        masterSafe.address,
+        bracketAddresses,
+        targetToken.address,
+        stableToken.address,
+        lowestLimit,
+        highestLimit,
+        currentPrice,
+        depositAmountStableToken,
+        depositAmountTargetToken
+      )
       await execTransaction(masterSafe, lw, batchTransaction)
-      // Close auction for deposits to be refelcted in exchange balance
+      // Close auction for deposits to be reflected in exchange balance
       await waitForNSeconds(301)
 
-      for (const bracketAddress of bracketAddresses.slice(0, fleetSize / 2)) {
-        let bracketExchangeBalance = (await exchange.getBalance(bracketAddress, stableToken.address)).toNumber()
-        assert.equal(bracketExchangeBalance, depositAmountStableToken)
-        bracketExchangeBalance = (await exchange.getBalance(bracketAddress, targetToken.address)).toNumber()
-        assert.equal(bracketExchangeBalance, 0)
-        const bracketPersonalTokenBalance = (await testToken.balanceOf(bracketAddress)).toNumber()
-        // This should always output 0 as the brackets should never directly hold funds
-        assert.equal(bracketPersonalTokenBalance, 0)
+      for (const bracketAddress of bracketAddresses) {
+        await checkCorrectnessOfDeposits(
+          currentPrice,
+          bracketAddress,
+          exchange,
+          stableToken,
+          targetToken,
+          depositAmountStableToken.div(new BN(2)),
+          depositAmountTargetToken.div(new BN(2))
+        )
       }
-      for (const bracketAddress of bracketAddresses.slice(fleetSize / 2 + 1, fleetSize / 2)) {
-        let bracketExchangeBalance = (await exchange.getBalance(bracketAddress, targetToken.address)).toNumber()
-        assert.equal(bracketExchangeBalance, depositAmountTargetToken)
-        bracketExchangeBalance = (await exchange.getBalance(bracketAddress, stableToken.address)).toNumber()
-        assert.equal(bracketExchangeBalance, 0)
-        const bracketPersonalTokenBalance = (await testToken.balanceOf(bracketAddress)).toNumber()
-        // This should always output 0 as the brackets should never directly hold funds
-        assert.equal(bracketPersonalTokenBalance, 0)
+    })
+    it("transfers tokens from fund account through trader accounts and into exchange via automatic deposit logic, p<1 && p>1", async () => {
+      const masterSafe = await deploySafe(gnosisSafeMasterCopy, proxyFactory, [lw.accounts[0], lw.accounts[1]], 2)
+      const fleetSize = 4
+      const lowestLimit = 0.8
+      const highestLimit = 1.2
+      const currentPrice = 1
```

Comment:
I'd use 0.9: in this way the price is indeed <1 and also you test the case where the price is not in the middle of the range.